spec/index.bs

```diff
@@ -1566,6 +1567,7 @@ dictionary GPUBindGroupLayoutBinding {
     GPUTextureComponentType textureComponentType = "float";
     boolean multisampled = false;
     boolean hasDynamicOffset = false;
+    GPUTextureFormat storageTextureFormat = {};
```

Comment:
Is it an error if `storageTextureFormat` doesn't match `textureComponentType`?

Reply:
In fact ```textureComponentType``` is not needed when ```storageTextureFormat``` is specified. Should I mention it clearly here?

Also should I add all the texture formats that are supported as writable storage texture here?

Reply:
Unfortunately it's not possible right now for `textureComponentType` to have no value. If it is not specified, then it will be `"float"`.

We could remove that default (so it defaults to "not set") and put spec text that says "if it's a sampled texture and `textureComponentType` is not set, textureComponentType defaults to `"float"`".

We really want to define an algebraic data type (/tagged union) here, but we have to work within the bounds of WebIDL. I think that's probably the best way to do it, unless WebIDL allows GPUBindGroupLayoutDescriptor to take `bindings` as a sequence of `(GPUBindGroupSampledTextureBinding or GPUBindGroupStorageTextureBinding or ...)` (which would be indistinguishable at the JS level except by the value of `type`). I don't think it does.

Reply:
Would be good to require the user to set `textureComponentType` explicitly for the sampled textures and `storageTextureFormat` for the storage ones. Defaulting to "float" wasn't something I was happy about :)

Reply:
OK with me!

Reply:
In the latest PR, I've removed the default value of ```textureComponentType```. I have also renamed ```textureComponentType``` to ```sampledTextureComponentType``` to indicate it is only related to sampled textures.